File: average_vector/average.py
def average():
    vector = []
    res = 0
    
    while True:
        verify = input("(y) if you want to input a number on array, (n) else: ")
        
        if verify.lower() == "y":
            num = (int(input("input some number on list: ")))
            vector.append(num)
            res += num
        elif verify.lower() == "n":
            break
        else:
            print("invalid option, please try again")
            
    if len(vector) != 0:
        res /= len(vector)
        print(f"the list is {vector}, the media is [{res}]")
    else:
        print("no number was inserted")  
        
average()

File: average_vector/average.js
const average = () => {
    let array = []
    let num = 0
    let res = 0

    while (true){
        let verify = prompt("(y) if you want to input a number on array, (n) else: ").toLowerCase()

        if (verify === "y"){
            num = (Number(prompt("input some number on array: ")))
            array.push(num)
            res += num
        }
        else if(verify === "n"){
            break
        }
        else{
            alert("invalid option, please try again")
        }
    }
     if(array.length > 0){
            res /= array.length
            alert(`the array [${array}], the average [${res}]`)
        }
    else{
            alert("no number was inserted")
        }
}

average()